eng/pipelines/mono/templates/build-job.yml

```diff
@@ -67,6 +67,9 @@ jobs:
         value: /p:OutputRid=linux-musl-${{ parameters.archType }}
       - name: _PortableBuild
         value: true
+    - ${{ if eq(parameters.osGroup, 'iOS') }}:
+      - name: osOverride
```

Comment:
There is already a _runtimeOSArg, why not use that, just like we do with wasm? https://github.com/dotnet/runtime/blob/deb35b83f33c27500940bb0313024f254974ae89/eng/pipelines/libraries/base-job.yml#L71